server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheContainerResource.java

```diff
@@ -386,7 +386,7 @@ public void registerChildren(ManagementResourceRegistration resourceRegistration
         super.registerChildren(resourceRegistration);
 
         // child resources
-        resourceRegistration.registerSubModel(new HealthResource(pathManager));
+        resourceRegistration.registerSubModel(new HealthResource(pathManager, runtimeRegistration));
```

Comment:
@slaskawi This still registers the HEALTH resource as a subModule, meaning that via DMR we can still do 
`/profile=clustered/subsystem=datagrid-infinispan/cache-container=clustered/health=HEALTH:read-resource() ` even though it will always return nothing. I think it would be better to surround L389 with an if and only register the subModule if runtimeRegistration is true. This way the erroneous health child will never appear in the DMR. WDYT?